Fix batcher processes and signals handling

GGP1 opened the pull request:
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/26086 |

## Description

Fixes some issues during the batcher process creation and signal handling.

## Tests


<details><summary>Starting API</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-comms-apid -f
2024/10/02 13:05:55 INFO: Starting API in foreground
2024/10/02 13:05:55 INFO: Generated private key file in /var/ossec/api/configuration/ssl/server.key
2024/10/02 13:05:55 INFO: Generated certificate file in /var/ossec/api/configuration/ssl/server.crt
```

</details>

<details><summary>Listing processes</summary>

```console
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  3152 ?        Ss   12:23   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root          37  0.0  0.0   4628  3568 pts/0    Ss   12:23   0:00 bash
root        7836  0.0  0.0   4628  3760 pts/1    Ss   12:39   0:00 bash
root       21082  0.0  0.0   2892  1000 pts/0    S+   13:05   0:00 /bin/sh /var/ossec/bin/wazuh-comms-apid -f
wazuh      21090 13.1  1.3 186144 112224 pts/0   S+   13:05   0:01 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      21131  1.1  1.1 405748 91188 pts/0    Sl+  13:05   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      21142  0.0  1.1 184572 89508 pts/0    S+   13:05   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      21144  2.0  1.1 184704 90760 pts/0    S+   13:05   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      21152  0.5  1.2 187712 99200 pts/0    S+   13:05   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      21153  0.4  1.2 187720 99136 pts/0    S+   13:05   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      21154  0.4  1.2 187712 99204 pts/0    S+   13:05   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      21155  0.4  1.2 187716 99204 pts/0    S+   13:05   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
root       21156  0.0  0.0   2792   996 ?        S    13:05   0:00 sleep 10
root       21200  0.0  0.0   7064  1560 pts/1    R+   13:06   0:00 ps aux
```

</details>

<details><summary>Shutdown</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-comms-apid -f
2024/10/02 13:05:55 INFO: Starting API in foreground
2024/10/02 13:05:55 INFO: Generated private key file in /var/ossec/api/configuration/ssl/server.key
2024/10/02 13:05:55 INFO: Generated certificate file in /var/ossec/api/configuration/ssl/server.crt
^C2024/10/02 13:06:12 INFO: Batcher pid 21144 - Received signal 2, initiating shutdown.
2024/10/02 13:06:13 INFO: Batcher pid 21144 - Received signal 15, initiating shutdown.
2024/10/02 13:06:13 INFO: MuxDemuxQueue pid 21142 - received signal SIGTERM
2024/10/02 13:06:13 INFO: MuxDemuxQueue pid 21142 - shutting down
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  3152 ?        Ss   12:23   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root          37  0.0  0.0   4628  3568 pts/0    Ss+  12:23   0:00 bash
root        7836  0.0  0.0   4628  3760 pts/1    Ss   12:39   0:00 bash
root       21327  0.0  0.0   2792  1008 ?        S    13:06   0:00 sleep 10
root       21368  0.0  0.0   7064  1580 pts/1    R+   13:06   0:00 ps aux
```

</details>

<details><summary>Same tests in background mode</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-comms-apid   
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  3152 ?        Ss   12:23   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root          37  0.0  0.0   4628  3556 pts/0    Ss   12:23   0:00 bash
root        7836  0.0  0.0   4628  3520 pts/1    Ss+  12:39   0:00 bash
root       60576  0.0  0.0   2792  1048 ?        S    14:27   0:00 sleep 10
wazuh      60578  1.5  1.1 184720 95008 ?        S    14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      60579  2.0  1.1 553228 91976 ?        Sl   14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      60596  0.0  1.0 184588 89044 ?        S    14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      60598  1.5  1.1 184588 90404 ?        S    14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      60604  2.0  1.2 187264 98508 ?        S    14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      60605  1.5  1.2 187268 98516 ?        S    14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      60606  2.0  1.2 187268 98516 ?        S    14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      60607  2.0  1.2 187268 98520 ?        S    14:27   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
root       60608  0.0  0.0   7064  1600 pts/0    R+   14:27   0:00 ps aux
root@wazuh-master:/# kill 60578
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  3152 ?        Ss   12:23   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root          37  0.0  0.0   4628  3568 pts/0    Ss   12:23   0:00 bash
root        7836  0.0  0.0   4628  3520 pts/1    Ss+  12:39   0:00 bash
root       60694  0.0  0.0   2792   988 ?        S    14:27   0:00 sleep 10
root       60695  0.0  0.0   7064  1604 pts/0    R+   14:27   0:00 ps aux
```

</details>

<details><summary>wazuh-control start</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-control start
2024/10/02 14:36:50 wazuh-db[64988] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/02 14:36:50 wazuh-db[64988] main.c:125 at main(): DEBUG: Wazuh home directory: /var/ossec
2024/10/02 14:36:50 wazuh-remoted[64994] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/02 14:36:50 wazuh-remoted[64994] main.c:130 at main(): DEBUG: Wazuh home directory: /var/ossec
2024/10/02 14:36:50 wazuh-remoted[64994] main.c:148 at main(): DEBUG: This is not a worker
2024/10/02 14:36:50 wazuh-modulesd:router: INFO: Loaded router module.
2024/10/02 14:36:50 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
Starting Wazuh v5.0.0...
Started wazuh-comms-apid...
Started wazuh-apid...
Started wazuh-csyslogd...
Started wazuh-dbd...
2024/10/02 14:36:53 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
Started wazuh-integratord...
Started wazuh-agentlessd...
2024/10/02 14:36:53 wazuh-authd[65140] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/02 14:36:53 wazuh-authd[65140] config.c:24 at authd_read_config(): DEBUG: Reading configuration 'etc/ossec.conf'
2024/10/02 14:36:53 wazuh-authd[65140] authd-config.c:276 at Read_Authd(): DEBUG: The tag <force><after_registration_time> is not defined. Applied default value: '3600'
2024/10/02 14:36:53 wazuh-authd[65140] authd-config.c:278 at Read_Authd(): DEBUG: The tag <force><key_mismatch> is not defined. Applied default value: 'true'
2024/10/02 14:36:53 wazuh-authd[65140] main-server.c:465 at main(): DEBUG: Wazuh home directory: /var/ossec
Started wazuh-authd...
2024/10/02 14:36:53 wazuh-db[65149] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/02 14:36:53 wazuh-db[65149] main.c:125 at main(): DEBUG: Wazuh home directory: /var/ossec
Started wazuh-db...
Started wazuh-execd...
Started wazuh-analysisd...
Started wazuh-syscheckd...
2024/10/02 14:36:54 wazuh-remoted[65214] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/02 14:36:54 wazuh-remoted[65214] main.c:130 at main(): DEBUG: Wazuh home directory: /var/ossec
2024/10/02 14:36:54 wazuh-remoted[65214] main.c:148 at main(): DEBUG: This is not a worker
Started wazuh-remoted...
Started wazuh-logcollector...
Started wazuh-monitord...
2024/10/02 14:36:55 wazuh-modulesd:router: INFO: Loaded router module.
2024/10/02 14:36:55 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
Started wazuh-modulesd...
Started wazuh-clusterd...
Completed.
```

</details>

<details><summary>wazuh-control stop</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-control stop
Killing wazuh-clusterd...
Killing wazuh-modulesd...
Killing wazuh-monitord...
Killing wazuh-logcollector...
Killing wazuh-remoted...
Killing wazuh-syscheckd...
Killing wazuh-analysisd...
wazuh-maild not running...
Killing wazuh-execd...
Killing wazuh-db...
Killing wazuh-authd...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
Killing wazuh-apid...
Killing wazuh-comms-apid...
Wazuh v5.0.0 Stopped
```

</details>

javiersanchz reviewed (dismissed):
## Review

The changes are correct.

The issues that arose with wazuh-comms-apid were resolved, and the environment works as expected without showing any errors

LGTM!

fdalmaup reviewed (changes requested):
Manually testing the branch, I have found the following:
<details>

```
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  1760 ?        Ss   09:58   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root        2326  0.0  0.0   4628  3832 pts/0    Ss   09:59   0:00 bash
wazuh      33104  0.1  0.5 184708 94972 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33105  5.0  0.5 553216 92480 ?        Sl   10:47   0:01 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33122  0.0  0.5 184576 89020 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33126  3.9  0.5 184708 90964 ?        S    10:47   0:01 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33130  0.2  0.6 187260 98784 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33131  0.3  0.6 187260 98780 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33132  0.3  0.6 187260 98784 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33133  0.2  0.6 187260 98784 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33155  8.3  0.7 816728 124368 ?       Sl   10:47   0:03 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      33156  0.0  0.5 176780 80512 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      33159  0.0  0.5 324244 80548 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      33162  0.0  0.5 471708 80676 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       33203  0.3  0.0 310776  8472 ?        Sl   10:47   0:00 /var/ossec/bin/wazuh-authd
wazuh      33223  0.4  0.1 988140 16904 ?        Sl   10:47   0:00 /var/ossec/bin/wazuh-db
root       33252  0.0  0.0  23832  3364 ?        Sl   10:47   0:00 /var/ossec/bin/wazuh-execd
wazuh      33263  0.9  0.1 3489584 29100 ?       Sl   10:47   0:00 /var/ossec/bin/wazuh-analysisd
root       33368  7.3  0.0 200732 12380 ?        SNl  10:47   0:02 /var/ossec/bin/wazuh-syscheckd
wazuh      33381  0.2  0.0 1210252 12324 ?       Sl   10:47   0:00 /var/ossec/bin/wazuh-remoted
root       33429  0.0  0.0 335316  7684 ?        Sl   10:47   0:00 /var/ossec/bin/wazuh-logcollector
wazuh      33445  0.0  0.0  23888  2892 ?        Sl   10:47   0:00 /var/ossec/bin/wazuh-monitord
root       33454  2.0  0.3 652968 60896 ?        Sl   10:47   0:00 /var/ossec/bin/wazuh-modulesd
wazuh      34254  1.2  0.5 320964 90728 ?        Sl   10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
wazuh      34256  0.0  0.4 173500 78936 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
wazuh      34257  0.0  0.4 173500 79056 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
root       34588  0.0  0.0   2792   988 ?        S    10:47   0:00 sleep 10
root       34654  0.0  0.0   7064  1552 pts/0    R+   10:48   0:00 ps aux
root@wazuh-master:/# /var/ossec/bin/wazuh-comms-apid -f 
wazuh-comms-apid: Orphan child process 33104 was terminated.
2024/10/03 10:48:08 INFO: Starting API in foreground
2024/10/03 10:48:08 ERROR: Connection in use: ('0.0.0.0', 27000)
2024/10/03 10:48:08 ERROR: Retrying in 1 second.
2024/10/03 10:48:09 ERROR: Connection in use: ('0.0.0.0', 27000)
2024/10/03 10:48:09 ERROR: Retrying in 1 second.
2024/10/03 10:48:10 ERROR: Connection in use: ('0.0.0.0', 27000)
2024/10/03 10:48:10 ERROR: Retrying in 1 second.
2024/10/03 10:48:11 ERROR: Connection in use: ('0.0.0.0', 27000)
2024/10/03 10:48:11 ERROR: Retrying in 1 second.
2024/10/03 10:48:12 ERROR: Connection in use: ('0.0.0.0', 27000)
2024/10/03 10:48:12 ERROR: Retrying in 1 second.
2024/10/03 10:48:13 ERROR: Can't connect to ('0.0.0.0', 27000)
2024/10/03 10:48:13 INFO: MuxDemuxQueue pid 34810 - received signal SIGTERM
2024/10/03 10:48:13 INFO: Batcher pid 34812 - Received signal 15, initiating shutdown.
2024/10/03 10:48:13 INFO: MuxDemuxQueue pid 34810 - shutting down
root@wazuh-master:/# /var/ossec/bin/wazuh-control stop   
Killing wazuh-clusterd...
Killing wazuh-modulesd...
Killing wazuh-monitord...
Killing wazuh-logcollector...
Killing wazuh-remoted...
Killing wazuh-syscheckd...
Killing wazuh-analysisd...
wazuh-maild not running...
Killing wazuh-execd...
Killing wazuh-db...
Killing wazuh-authd...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
Killing wazuh-apid...
wazuh-comms-apid not running...
Wazuh v5.0.0 Stopped
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  1760 ?        Ss   09:58   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root        2326  0.0  0.0   4628  3832 pts/0    Ss   09:59   0:00 bash
wazuh      33105  5.0  0.5 553216 92480 ?        Sl   10:47   0:03 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33122  0.0  0.5 184576 89020 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33126  4.0  0.5 184708 90964 ?        S    10:47   0:02 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33130  0.2  0.6 187260 98784 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33131  0.2  0.6 187260 98780 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33132  0.2  0.6 187260 98784 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      33133  0.2  0.6 187260 99000 ?        S    10:47   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
root       35101  0.0  0.0   2792  1000 ?        S    10:48   0:00 sleep 10
root       35163  0.0  0.0   7064  1596 pts/0    R+   10:48   0:00 ps aux
```

</details>

The Comms API processes are not being shut down.

GGP1 commented:
> Manually testing the branch, I have found the following:
> 
> The Comms API processes are not being shut down.

This was caused by the fact that killing the parent process during the pid files cleanup did not kill the child processes. I fixed this by killing the process group instead (parent and children always share the same group).

I also modified some of the logs to make them cleaner.

### Tests

<details><summary>wazuh-control start</summary>

```console
2024/10/03 13:14:01 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
Starting Wazuh v5.0.0...
Started wazuh-comms-apid...
Started wazuh-apid...
Started wazuh-csyslogd...
Started wazuh-dbd...
2024/10/03 13:14:04 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
Started wazuh-integratord...
Started wazuh-agentlessd...
2024/10/03 13:14:04 wazuh-authd[23208] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/03 13:14:04 wazuh-authd[23208] config.c:24 at authd_read_config(): DEBUG: Reading configuration 'etc/ossec.conf'
2024/10/03 13:14:04 wazuh-authd[23208] authd-config.c:276 at Read_Authd(): DEBUG: The tag <force><after_registration_time> is not defined. Applied default value: '3600'
2024/10/03 13:14:04 wazuh-authd[23208] authd-config.c:278 at Read_Authd(): DEBUG: The tag <force><key_mismatch> is not defined. Applied default value: 'true'
2024/10/03 13:14:04 wazuh-authd[23208] main-server.c:465 at main(): DEBUG: Wazuh home directory: /var/ossec
Started wazuh-authd...
2024/10/03 13:14:04 wazuh-db[23221] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/03 13:14:04 wazuh-db[23221] main.c:125 at main(): DEBUG: Wazuh home directory: /var/ossec
Started wazuh-db...
Started wazuh-execd...
Started wazuh-analysisd...
Started wazuh-syscheckd...
2024/10/03 13:14:04 wazuh-remoted[23263] debug_op.c:116 at _log_function(): DEBUG: Logging module auto-initialized
2024/10/03 13:14:04 wazuh-remoted[23263] main.c:130 at main(): DEBUG: Wazuh home directory: /var/ossec
2024/10/03 13:14:04 wazuh-remoted[23263] main.c:148 at main(): DEBUG: This is not a worker
Started wazuh-remoted...
Started wazuh-logcollector...
Started wazuh-monitord...
2024/10/03 13:14:05 wazuh-modulesd:router: INFO: Loaded router module.
2024/10/03 13:14:05 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
Started wazuh-modulesd...
Started wazuh-clusterd...
Completed.
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  1492 ?        Ss   12:53   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root        1807  0.0  0.0   4628  2812 pts/0    Ss   12:54   0:00 bash
root       17349  0.0  0.0   4628  3428 pts/1    Ss+  13:07   0:00 bash
wazuh      23116  0.4  0.9 184712 78772 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23117  2.2  0.9 553220 73716 ?        Sl   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23134  0.0  0.8 184580 69452 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23136  1.5  0.8 184580 72552 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23142  0.6  1.0 187260 83276 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23143  0.5  1.0 187260 83276 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23144  0.5  1.0 187260 83284 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23145  0.5  1.0 187260 83284 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py
wazuh      23162 26.6  1.5 816732 124428 ?       Sl   13:14   0:02 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      23163  0.0  0.9 176780 80464 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      23166  0.0  0.9 324244 80564 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      23169  0.0  0.9 471708 80628 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       23210  0.3  0.0 310816  7324 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-authd
wazuh      23223  2.2  0.2 988188 17720 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-db
root       23233  0.0  0.0  23892  3216 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-execd
wazuh      23244  5.6  0.5 4439996 41048 ?       Sl   13:14   0:00 /var/ossec/bin/wazuh-analysisd
root       23253 19.2  0.1 200808 11856 ?        SNl  13:14   0:01 /var/ossec/bin/wazuh-syscheckd
wazuh      23267  0.1  0.2 1210232 18832 ?       Sl   13:14   0:00 /var/ossec/bin/wazuh-remoted
root       23436  0.0  0.0 335392  7440 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-logcollector
wazuh      23459  0.0  0.0  15668  2876 ?        S    13:14   0:00 /var/ossec/bin/wazuh-monitord
root       23468 20.4  1.3 777396 106572 ?       Sl   13:14   0:01 /var/ossec/bin/wazuh-modulesd
wazuh      24266  5.5  1.1 320984 91096 ?        Sl   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
wazuh      24291  0.0  0.9 173520 79060 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
wazuh      24292  0.0  0.9 173520 79000 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
root       24322  0.0  0.0   2792  1020 ?        S    13:14   0:00 sleep 10
root       24552  0.0  0.0   7064  1556 pts/0    R+   13:14   0:00 ps aux
```

</details>

<details><summary>Start the API in foreground mode</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-comms-apid -f
wazuh-comms-apid: Orphan child process 23116 was terminated.
2024/10/03 13:14:24 INFO: Starting API in foreground
2024/10/03 13:14:24 INFO: Listening on 0.0.0.0:27000
```

</details>

<details><summary>List processes</summary>

```console
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  1492 ?        Ss   12:53   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root        1807  0.0  0.0   4628  2812 pts/0    Ss   12:54   0:00 bash
root       17349  0.0  0.0   4628  3448 pts/1    Ss   13:07   0:00 bash
wazuh      23162  2.4  1.5 816732 124428 ?       Sl   13:14   0:02 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      23163  0.0  0.9 176780 80464 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      23166  0.0  0.9 324244 80564 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      23169  0.0  0.9 471708 80628 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       23210  0.4  0.0 310816  7324 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-authd
wazuh      23223  0.2  0.2 988188 18364 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-db
root       23233  0.0  0.0  23892  3216 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-execd
wazuh      23244  0.4  0.5 4702140 41048 ?       Sl   13:14   0:00 /var/ossec/bin/wazuh-analysisd
root       23253  2.2  0.1 200808 11980 ?        SNl  13:14   0:02 /var/ossec/bin/wazuh-syscheckd
wazuh      23267  0.0  0.2 1210232 20880 ?       Sl   13:14   0:00 /var/ossec/bin/wazuh-remoted
root       23436  0.0  0.0 400928  7440 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-logcollector
wazuh      23459  0.0  0.0  23864  2876 ?        Sl   13:14   0:00 /var/ossec/bin/wazuh-monitord
root       23468  1.4  1.3 777396 107680 ?       Sl   13:14   0:01 /var/ossec/bin/wazuh-modulesd
wazuh      24266  0.3  1.1 320984 91100 ?        Sl   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
wazuh      24291  0.0  0.9 173520 79132 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
wazuh      24292  0.0  0.9 173520 79004 ?        S    13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/framework/scripts/wazuh_clusterd.py
root       25048  0.0  0.0   2892   976 pts/0    S+   13:14   0:00 /bin/sh /var/ossec/bin/wazuh-comms-apid -f
wazuh      25056  1.2  1.3 184700 110116 pts/0   S+   13:14   0:01 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      25057  2.8  1.1 553208 92456 pts/0    Sl+  13:14   0:02 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      25074  0.0  1.1 184568 89556 pts/0    S+   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      25076  1.9  1.1 184700 90860 pts/0    S+   13:14   0:01 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      25082  0.1  1.2 187260 98920 pts/0    S+   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      25083  0.1  1.2 187260 98924 pts/0    S+   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      25084  0.1  1.2 187260 98928 pts/0    S+   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
wazuh      25085  0.1  1.2 187260 98924 pts/0    S+   13:14   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/apis/scripts/wazuh_comms_apid.py -f
root       26056  0.0  0.0   2792  1008 ?        S    13:15   0:00 sleep 10
root       26121  0.0  0.0   7064  1584 pts/1    R+   13:15   0:00 ps aux
```

</details>

<details><summary>wazuh-control stop</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-control stop
Killing wazuh-clusterd...
Killing wazuh-modulesd...
Killing wazuh-monitord...
Killing wazuh-logcollector...
Killing wazuh-remoted...
Killing wazuh-syscheckd...
Killing wazuh-analysisd...
wazuh-maild not running...
Killing wazuh-execd...
Killing wazuh-db...
Killing wazuh-authd...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
Killing wazuh-apid...
Killing wazuh-comms-apid...
Wazuh v5.0.0 Stopped
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  1492 ?        Ss   12:53   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root        1807  0.0  0.0   4628  2812 pts/0    Ss+  12:54   0:00 bash
root       17349  0.0  0.0   4628  3480 pts/1    Ss   13:07   0:00 bash
root       27939  0.0  0.0   2792  1024 ?        S    13:17   0:00 sleep 10
root       27980  0.0  0.0   7064  1576 pts/1    R+   13:18   0:00 ps aux
```

</details>

<details><summary>Shutdown in foreground mode</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-comms-apid -f
2024/10/03 13:57:00 INFO: Starting API in foreground
2024/10/03 13:57:00 INFO: Listening on 0.0.0.0:27000
^C2024/10/03 13:57:00 INFO: Shutting down
2024/10/03 13:59:02 INFO: Batcher (pid: 60788) - Received signal SIGTERM, shutting down
2024/10/03 13:59:03 INFO: MuxDemuxQueue (pid: 60787) - Received signal SIGTERM, shutting down
root@wazuh-master:/# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   4364  1488 ?        Ss   12:53   0:00 bash /scripts/entrypoint.sh wazuh-master master-node master
root        1807  0.0  0.0   4628  2824 pts/0    Ss   12:54   0:00 bash
root       17349  0.0  0.0   4628  3476 pts/1    Ss+  13:07   0:00 bash
root       62535  0.0  0.0   2792  1108 ?        S    13:59   0:00 sleep 10
root       62660  0.0  0.0   7064  1660 pts/0    R+   13:59   0:00 ps aux
```

</details>
